modules/services/crashlogging/src/main/java/au/com/shiftyjelly/pocketcasts/crashlogging/PocketCastsCrashLoggingDataProvider.kt

```diff
@@ -37,22 +36,18 @@ class PocketCastsCrashLoggingDataProvider @Inject constructor(
 
     override val sentryDSN: String = BuildConfig.SENTRY_DSN
 
-    override val user: Flow<CrashLoggingUser?> = settings.linkCrashReportsToUser.flow.map { linkCrashReportsToUser ->
-        if (linkCrashReportsToUser) {
-            syncManager.getEmail()?.let { email ->
-                CrashLoggingUser(
-                    email = email,
-                    userID = "",
-                    username = "",
-                )
-            }
-        } else {
-            null
+    override val user: Flow<CrashLoggingUser?> = observeUser.invoke().map { userMail ->
+        userMail?.let {
+            CrashLoggingUser(
+                email = it.email,
+                userID = "",
+                username = "",
+            )
         }
     }
 
     override fun crashLoggingEnabled(): Boolean {
-        return settings.sendCrashReports.value
+        return crashReportPermissionCheck.invoke()
```

Comment:
❓ Should we disable crash logging in debug?
```suggestion
        return !BuildConfig.DEBUG && crashReportPermissionCheck.invoke()
```

Reply:
Thanks it would be good to remove the Sentry exceptions in debug. We don't need the extra Sentry events.

Reply:
I think we could just drop the report a crash in our developer menu. But happy to do which ever you think is better.

Reply:
Disabling Sentry in debug builds sounds good to me. But having crash reporting in developer menu is useful. We used it, for example, in #1778 and #1776. It might be weird to have a setting that doesn't work in debug builds so we might add it conditionally if it is a release build. Or we can add a description to the setting that it works only in release builds.

https://github.com/Automattic/pocket-casts-android/blob/df1a9b82e755d88b8d1dee068d73bc87c65b8cd1/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/developer/DeveloperPage.kt#L123-L125

Reply:
I see @MiSikora , thanks for the input. I'll look for some way to disable Sentry in debug and keep the "Report a crash" features.

> Or we can add a description to the setting that it works only in release builds.

You mean to the `DeveloperPage`? Correct me if I'm wrong, but if I see correctly, this screen is available only in debug

Reply:
btw! I'll go ahead and merge this PR to unblock further progress but it's only feature branch PR.

Reply:
> You mean to the `DeveloperPage`? Correct me if I'm wrong, but if I see correctly, this screen is available only in debug

Yes, but for testing purposes we can produce a release build with this page visible.